src/goto-programs/abstract-interpretation/interval_domain.h

```diff
@@ -26,6 +23,17 @@ using real_intervalt =
 class interval_domaint : public ai_domain_baset
 {
 public:
+  using integer_intervalt = interval_templatet<BigInt>;
+  using real_intervalt =
+    interval_templatet<boost::multiprecision::cpp_bin_float_100>;
+
+  using interval = std::variant<
+    std::shared_ptr<integer_intervalt>,
+    std::shared_ptr<real_intervalt>,
+    std::shared_ptr<wrapped_interval>>;
+
+  using interval_map = std::unordered_map<irep_idt, interval, irep_id_hash>;
```

Comment:
Could you please add a comment here saying what "no entry" in this map means, and also what an entry means, where the associated shared_ptr points to nothing (if that can even happen)? Just so we don't forget.

Reply:
I added the comment

Reply:
Can "key exists, but shared-pointer is null" ever happen or is it supposed to happen?

Reply:
Added a new note